File: src/main/java/org/dedda/bratwurst/parse/ExitParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.Exit;

import static org.dedda.bratwurst.parse.Patterns.END;

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
public class ExitParser {

    public Exit parse(String line, int lineNumber) {
        return line.matches(END) ? new Exit(lineNumber) : null;
    }

}

File: src/main/java/org/dedda/bratwurst/parse/ExitParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.Exit

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
class ExitParser : InstructionParser() {
    override fun parse(line: String, lineNumber: Int): Exit? {
        return if (line.matches(Regex(Patterns.END))) Exit(lineNumber) else null
    }
}